pkg/resource/deploy/source_eval.go

```diff
@@ -1422,6 +1433,22 @@ func (rm *resmon) RegisterResource(ctx context.Context,
 	}
 
 	additionalSecretOutputs := req.GetAdditionalSecretOutputs()
+	if rm.loader != nil {
+		// try to load the package reference and infer which properties are supposed to be secret.
+		var version *semver.Version
+		if parsedVersion, err := semver.Parse(req.GetVersion()); err == nil {
+			version = &parsedVersion
+		}
+		if pkgReference, err := rm.loader.LoadPackageReference(t.Package().String(), version); err == nil {
```

Comment:
Is LoadPackageReference definitely cached? We really don't want to incur re-load penalties per resource here.

Reply:
The default loader which is a `pluginLoader` does caching by default so a loaded package reference shouldn't need to be reloaded every time.